cellxgene_schema_cli/scripts/ontology_processing.py

```diff
@@ -0,0 +1,162 @@
+    output_json_file: str = env.PARSED_ONTOLOGIES_FILE,
+):
+
+    """
+    Parser all owl files in working_dir. Extracts information from all classes in the owl file.
+    The extracted information is written into a gzipped a json file with the following structure:
+    {
+        "ontology_name":
+            {
+            "term_id": {
+                "label": "..."
+                "ancestors": [
+                    "ancestor1_term_id_1",
+                    "ancestor2_term_id_2"
+                    ]
+                }
+            }
+
+            "term_id2": {
+                ...
+            }
+
+            ...
+            }
+    }
+
+    :param str working_dir: path to folder with owl files
+    :param str owl_info_yml: path to writable directory where owl files will be downloaded to
+    :param str owl_info_yml: path to yaml file wit owl information
+    :param str output_json_file: path to output jsaon file
+
+    :rtype None
+    """
+
+    with open(owl_info_yml, "r") as owl_info_handle:
+        owl_info = yaml.safe_load(owl_info_handle)
+
+    owl_files = [
+        os.path.join(working_dir, i)
+        for i in os.listdir(working_dir)
+        if i.endswith(".owl")
+    ]
+
+    # Parse owl files
+    onto_dict = {}
+    for i in owl_files:
+        onto = owlready2.get_ontology(i).load()
+        onto_dict[onto.name] = {}
+
+        print(f"Processing {onto.name}")
+        for onto_class in onto.classes():
+
+            term_id = onto_class.name.replace("_", ":")
+
+            # Skip terms that are not direct children from this ontology
+            if not onto.name == term_id.split(":")[0]:
+                continue
+
+            # If there are specified target terms then only work with them
+            if onto.name in owl_info:
+                if "only" in owl_info[onto.name]:
+                    if term_id not in owl_info[onto.name]["only"]:
+                        continue
+
+            onto_dict[onto.name][term_id] = dict()
+            try:
+                onto_dict[onto.name][term_id]["label"] = onto_class.label[0]
+            except IndexError:
+                onto_dict[onto.name][term_id]["label"] = ""
+
+            if onto.name == "NCBITaxon":
+                onto_dict[onto.name][term_id]["ancestors"] = []
+            else:
+                onto_dict[onto.name][term_id]["ancestors"] = _get_ancestors(onto_class, onto.name)
+
+    with gzip.open(output_json_file, "wt") as output_json:
+        json.dump(onto_dict, output_json, indent=2)
+
+
+def _get_ancestors(onto_class: owlready2.entity.ThingClass, ontololgy_name: str):
+    """
+    Returns a list of ancestors ids of the given onto class, only returns those belonging to ontology_name,
+    it will format the id from the form CL_xxxx to CL:xxxx
+
+    :param owlready2.entity.ThingClass onto_class: the class for which ancestors will be retrieved
+    :param str ontololgy_name: only ancestors from this ontology will be kept
+    """
+    return [
```

Comment:
Can you refactor this for readability/clarity. Ideally avoid using single character variable names